mars/services/task/execution/api.py

```diff
@@ -89,3 +102,28 @@ def get_stage_processors(self):
 
 def register_executor_cls(executor_cls: Type[TaskExecutor]):
     _name_to_task_executor_cls[executor_cls.name] = executor_cls
+
+
+class Fetcher:
+    name = None
+
+    @abstractmethod
+    def __init__(self, meta: Dict, **kwargs):
+        pass
+
+    @abstractmethod
+    async def get(self, conditions: List = None):
+        pass
+
+    @classmethod
+    def create(cls, meta: Dict, **kwargs) -> "Fetcher":
+        name = meta["fetcher"]
```

Comment:
Do we need to store `fetcher` in meta for every chunk?

Reply:
The fetcher implementation has been refined.